src/actions/device.ts

```diff
@@ -70,48 +72,90 @@ export const sendListToRemoteInport = (name: string, values: number[]): AppThunk
-	() => {
-		const value = name;
-		const message = {
-			address: "/rnbo/inst/0/presets/load",
-			args: [
-				{ type: "s", value }
-			]
-		};
-		oscQueryBridge.sendPacket(writePacket(message));
+export const loadPresetOnRemote = (preset: PresetRecord): AppThunk =>
+	(dispatch) => {
+		try {
+			const message = {
+				address: "/rnbo/inst/0/presets/load",
+				args: [
+					{ type: "s", value: preset.name }
+				]
+			};
+			oscQueryBridge.sendPacket(writePacket(message));
+			dispatch(showNotification({
+				level: NotificationLevel.success,
+				title: "Preset loaded",
+				message: `Preset "${preset.name} has been loaded.`
+			}));
+		} catch (err) {
+			dispatch(showNotification({
+				level: NotificationLevel.error,
+				title: `Error while trying to load preset ${preset.name}`,
+				message: "Please check the consolor for further details."
+			}));
+			console.log(err);
+		}
 	};
 
 export const savePresetToRemote = (name: string): AppThunk =>
-	() => {
-		const value = name;
-		const message = {
-			address: "/rnbo/inst/0/presets/save",
-			args: [
-				{ type: "s", value }
-			]
-		};
-		oscQueryBridge.sendPacket(writePacket(message));
-	};
-
-export const loadPatcher = (name: string, inst: number = 0): AppThunk =>
-	() => {
-		let message = {
-			address: "/rnbo/inst/control/load",
-			args: [
-				{ type: "i", value: inst },
-				{ type: "s", value: name }
-			]
-		};
-		if (name === UNLOAD_PATCHER_NAME) {
-			message = {
-				address: "/rnbo/inst/control/unload",
+	(dispatch) => {
+		try {
+			const message = {
+				address: "/rnbo/inst/0/presets/save",
 				args: [
-					{ type: "i", value: inst }
+					{ type: "s", value: name }
 				]
 			};
+			oscQueryBridge.sendPacket(writePacket(message));
+			dispatch(showNotification({
+				level: NotificationLevel.success,
+				title: "Preset saved",
+				message: `Preset "${name} has been saved.`
+			}));
+		} catch (err) {
+			dispatch(showNotification({
+				level: NotificationLevel.error,
+				title: `Error while trying to save preset ${name}`,
+				message: "Please check the consolor for further details."
+			}));
+			console.log(err);
+		}
+	};
+
+export const loadPatcher = (patcher: PatcherRecord, inst: number = 0): AppThunk =>
+	(dispatch) => {
+		try {
+			let message;
+			if (patcher.name === UNLOAD_PATCHER_NAME) {
+				message = {
+					address: "/rnbo/inst/control/unload",
+					args: [
+						{ type: "i", value: inst }
+					]
+				};
+			} else {
+				message = {
+					address: "/rnbo/inst/control/load",
+					args: [
+						{ type: "i", value: inst },
+						{ type: "s", value: patcher.name }
+					]
+				};
+			}
+			oscQueryBridge.sendPacket(writePacket(message));
+			dispatch(showNotification({
+				level: NotificationLevel.success,
+				title: patcher.name === UNLOAD_PATCHER_NAME ?  "Unloaded patcher" : "Loaded patcher",
+				message: patcher.name === UNLOAD_PATCHER_NAME ? "" : `The patcher ${patcher.name} has been loaded`
+			}));
```

Comment:
at this point you don't actually know that the patcher has been loaded or unloaded as it may take some time or fail.. you can check the OSC response to see though, or you could just change the message and say something like `The patcher X load has been sent`

Reply:
yeah maybe we just take it out and only leave the client side error in? Given the potential changes it seems like implementing transactions for this at this point might be a stretch or become redundant soonish?!

Reply:
yeah, you could do the same with presets